controllers/storagecluster/cephrbdmirrors.go

```diff
@@ -124,3 +130,283 @@ func (obj *ocsCephRbdMirrors) ensureDeleted(r *StorageClusterReconciler, sc *ocs
 	}
 	return reconcile.Result{}, r.deleteCephRbdMirrorInstance(cephRbdMirrors)
 }
+
+func (r *StorageClusterReconciler) ensureRbdMirrorDebug(sc *ocsv1.StorageCluster) error {
+	job := &batchv1.Job{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{
+		Name:      "rbd-mirror-debug",
+		Namespace: sc.Namespace,
+	}, job)
+
+	if sc.Spec.Mirroring.Enabled {
+		if err != nil {
+			if errors.IsNotFound(err) {
+				// Create the debug job
+				return r.Client.Create(context.TODO(), rbdMirrorDebugJob(sc))
+			}
+			return err
+		}
+		return nil
+	} else if err == nil {
+		// mirroring is not enabled but the job exists
+		err := r.Client.Delete(context.TODO(), job)
+		if err != nil {
+			return err
+		}
+		// Create a undo job to remove the ceph configs applied by the debug job
+		err = r.Client.Create(context.TODO(), undoRbdMirrorDebugJob(sc))
+		if err != nil {
+			return err
+		}
+		// wait for the undo job to succeed & then delete it
+		err = utilwait.PollImmediate(5*time.Second, 60*time.Second, func() (done bool, err error) {
```

Comment:
What if this job takes more time complete, like an unstable network case of slow image pulling etc?

Reply:
So you are suggesting I should increase the timeout? to probably 120 sec/ 180 sec/ 300 sec?

Reply:
If it fails for 60 seconds and we are going to retry as Reconcile fails, we are good. If not it should be taken care of.

Reply:
Yes, we will retry as reconcile will fail & would be tried again